vt_graph_api/graph.py

```diff
@@ -1805,17 +2000,68 @@ def create_collection(self, name=None, description=None):
         collection_id=collection_id
     )
 
-  def set_representation(self, representation):
-    """Sets Graph representation.
 
-    Args:
-      representation: Graph representation. See :py:class::`RepresentationType`.
+  def _generate_group_node_id(self, nodes_id):
+    return ",".join(nodes_id).replace("=", "")
+
+
+  def create_group(self, node_ids, group_name):
     """
-    self.representation = representation
 
+    Args:
+      node_ids:
+      group_name:
 
+    Returns:
 
+    Raises: CreateGroupError if the
 
+    """
+    group_node_id = self._generate_group_node_id(node_ids)
+    node_ids_set = set(node_ids)
+
+    # Check if the user has provided node ids
+    if not node_ids:
+      raise vt_graph_api.errors.CreateGroupError(
+          "A group must contain at least one node.")
+
+    # Check if all the nodes exists.
+    for node_id in node_ids_set:
+      if node_id not in self.nodes:
+        raise vt_graph_api.errors.CreateGroupError(
+            "Node {node_id} is not in the Graph.".format(
+                node_id=node_id))
+
+    # Check if nodes are already in a group.
+    nodes_already_in_a_group = set()
+    for group_node in self.group_nodes.values():
+      for node_id in group_node["entity_attributes"]["grouped_node_ids"]:
+        nodes_already_in_a_group.add(node_id)
+
+    intersection = nodes_already_in_a_group.intersection(node_ids)
```

Comment:
given the cardinality if the nodes to add to a group will be usually lower than nodes in current groups, what about checking if the nodes in other groups are already in node_ids (may be a set) and raise the error as soon as possible, at line 2039?